mango-core/src/main/java/org/calrissian/mango/domain/BaseAttributeStore.java

```diff
@@ -31,22 +31,21 @@
  */
 public class BaseAttributeStore implements AttributeStore {
 
-    private Multimap<String, Attribute> attributes = ArrayListMultimap.create();
+    protected Multimap<String, Attribute> attributes;
 
     protected BaseAttributeStore(Multimap<String, Attribute> attributes) {
+        checkNotNull(attributes);
         this.attributes = attributes;
     }
 
-    @Deprecated
-    public void put(Attribute attribute) {
+    protected void put(Attribute attribute) {
```

Comment:
Why leave the put methods.  It just invites subclasses to have mutable capabilities which defeats the guaranteed benefits of having an immutable data object.